Automattic/newspack-plugin
# Add initial GAM integration

### All Submissions:






### Changes proposed in this Pull Request:



Sets up Google Ad Manager integration. For now, just fetching the network code, which currently has to be supplied by the user.
This integration is reusing Site Kit's authentication, but with additional permission scope for GAM. **This is not yet supported by Site Kit unless used in [development mode](https://sitekit.withgoogle.com/documentation/using-site-kit-on-a-staging-environment/) (with a custom OAuth2 app).**

### How to test the changes in this Pull Request:

All steps to be performed in the Advertising Wizard, Google Ad Manager view, Global Code tab. In all cases, if the site has a user-supplied network code, it should be displayed (but not editable). 

1. Happy path: use a fresh site with Newspack set up (using the version from this branch) and the user having access to a GAM account
    1. Deactivate the Site Kit plugin, observe a notice prompting to activate it
    2. Activate Site Kit, observe information about missing GAM data and a "Refresh GAM data" link
    3. Click the link, observe the UI displaying the email address of the user who authenticated with Site Kit and their GAM network code
    4. Ensure the ads from that network are served after the ads have been configured
1. Legacy path: use a site connected to Site Kit while a previous (`master`) version of Newspack was active
    1. Observe information about insufficient permission scopes
    2. Redo Site Kit setup as prompted in Site Kit dashboard
    3. Observe pts. 3,4 same as above
1. Other paths
    1. Set up a Newspack site as a user with no GAM account. Observe an error notice stating that this feature requires a GAM account.  

### Other information:
